frontend/src/Frontend/Examples/ScreenKeyboard/Main.hs

```diff
@@ -50,8 +52,24 @@ inputW buttonE = do
     let
       html = _inputElement_raw input     -- html element
       cur = current $ value input        -- actual string
-    input <- inputElement $ def
-      & inputElementConfig_setValue .~ fmap snd newStringE
+      cfg = def & inputElementConfig_setValue .~ fmap snd newStringE
+
+    input <- inputElement cfg
+    -- Use this to trigger an event *after* the value has been updated.
+    valueChangedBySetValue :: Event t String <-
+      case _inputElementConfig_setValue cfg of
+        Nothing -> return never
+        Just eSetValue -> requestDomAction $ getValue html <$ eSetValue
+
+    -- Keep focus on the HTML element and maintain our last known cursor
+    -- position.
+    latestPos <- holdDyn 0 $ fmap fst newStringE
+    latestVal <- holdDyn "" valueChangedBySetValue
+    void $ performArg (\(n, _) -> do
+      focus html
+      setSelectionStart html n
+      setSelectionEnd html n) (updated $ zipDynWith (,) latestPos latestVal)
```

Comment:
When tracing, I see the nested function gets triggered twice since `latestPos` gets updated and `latestVal` gets updated independently. What's the preferred way to "de-dup" in such a situation? Or better yet, only trigger this inner function after `latestVal` changes and extract the value from `latestPos` on request (e.g. should I adjust constraints to allow `sample`-ing?).